File: C# (ASP.NET & WPF)/C# - INSY7315 - EliteRentals - (Web App, Mobile App & API)/Elite-Rentals-Rest-API/EliteRentalsAPI/Models/Maintenance.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EliteRentalsAPI.Models
{
    public class Maintenance
    {
        [Key]
        public int MaintenanceId { get; set; }

        [ForeignKey("Tenant")]
        public int TenantId { get; set; }
        public User? Tenant { get; set; }

        [ForeignKey("Property")]
        public int PropertyId { get; set; }
        public Property? Property { get; set; }

        [ForeignKey("Caretaker")]
        public int? AssignedCaretakerId { get; set; }
        public User? Caretaker { get; set; }

        public string Description { get; set; } = "";
        public string Category { get; set; } = "";
        public string Urgency { get; set; } = "Low";
        public string Status { get; set; } = "Pending";
        public byte[]? ProofData { get; set; }
        public string? ProofType { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}

File: Kotlin/Kotlin - PROG7314 - EliteRentals (Mobile App & API Development)/Elite-Rentals-Mobile-App/app/src/main/java/com/rentals/eliterentals/Maintenance.kt
package com.rentals.eliterentals

data class Maintenance(
    val maintenanceId: Int,
    val tenantId: Int,
    val propertyId: Int,
    val description: String,
    val category: String?,
    val urgency: String?,
    var status: String,
    val createdAt: String,
    val updatedAt: String?,
    val imageUrl: String?,
    val assignedCaretakerId: Int?
)
